substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/DebugHelper.java

```diff
@@ -281,6 +295,493 @@ public static void printLocationInfo(@SuppressWarnings("unused") IsolateThread t
             SubstrateDiagnostics.printLocationInfo(Log.log(), mem, true, true);
         }
     }
+    public static class IdeDebugHelper {
+        /**
+         * Convert instance of java.lang.String to utf8 c-string.
+         * @param thread
+         * @param ptr pointer to instance of java.lang.String
+         * @return c-string
+         */
+        @CEntryPoint(name = "svm_dbg_string_to_utf8", include = IncludeDebugHelperMethods.class, publishAs = Publish.SymbolOnly)
+        @CEntryPointOptions(prologue = SetThreadAndHeapBasePrologue.class)
+        public static CCharPointer stringToUtf8(@SuppressWarnings("unused") IsolateThread thread, Pointer ptr) {
+            Object obj = ptr.toObject();
+            return CTypeConversion.toCString((String) obj).get();
```

Comment:
This leaks memory because `CTypeConversion.toCString(...)` allocates a `CCharPointerHolder` that contains the pinned object. You would need to call `CCharPointerHolder.close()` once you don't need the native memory anymore.